docs/_docs/thin-clients/java-thin-client.adoc

```diff
@@ -170,19 +181,96 @@ NOTE: The `getAll()` method retrieves the results from the cursor and closes it.
 
 Read more about using `SqlFieldsQuery` and SQL API in the link:SQL/sql-api[Using SQL API] section.
 
-== Handling Exceptions
+== Cluster API
 
-=== Handling Node Failures
+Cluster functionality (APIs for accessing cluster and nodes) is provided via the `ClientCluster` interface. You can get an instance of `ClientCluster` from `IgniteClient` as follows:
+[source, java]
+-------------------------------------------------------------------------------
+include::{sourceCodeFile}[tag=client-cluster,indent=0]
+-------------------------------------------------------------------------------
 
-When you provide the addresses of multiple nodes in the client configuration, the client automatically switches to the next node if the current connection fails and retries any ongoing operation.
+Through the `ClientCluster` interface you can:
+
+* Get or set cluster state
+* Get a list of all cluster members
+* Create logical Cluster Groups
 
-In the case of atomic operations, failover to another node is transparent to the user.  However, if you execute a scan query or a SELECT query, it may return duplicate results. This can happen because queries return data in pages, and if the node that the client is connected to goes down while the client retrieves the pages, the client connects to another node and executes the query again. To avoid this, you have to write some code in your application that checks if the entries returned by the client are duplicated. Consider the following code:
+== Cluster groups
```

Comment:
I think this section should be a subsection of "Cluster API" and should give an example of how to get a group of nodes for a specific purpose. Also, "Cluster groups" is an incorrect term. I know it's used in the API itself, but it should be "Node groups".  In fact, you don't have to use the term, you can simple say something like "here is how you can group nodes based on some characteristic (server nodes, clients, attributes, etc.)." and give an example.